UIforETW/ChildProcess.cpp

```diff
@@ -112,10 +111,10 @@ DWORD ChildProcess::ListenerThread()
 	return 0;
 }
 
-
+_Pre_satisfies_(!(this->hProcess_))
 bool ChildProcess::Run(bool showCommand, std::wstring args)
 {
-	assert(!hProcess_);
+	ATLASSERT(!hProcess_);

```

Comment:
Some versions of assert bring up a dialog box to let the user choose what to do. This can be quite brilliant, but if the dialog box is naively invoked on the current thread when it will run a message pump on the current thread, which affects behavior and may even allow execution to flow back to the assert location (i.e.; if you have an assert in WM_PAINT or WM_TIMER).

The fix is to open the dialog on an assert-specific thread (which is a bit of engineering work) or not open a dialog.

Anyway, I'm not convinced by my own UIETWASSERT argument since it is a bit ugly and non-standard and solves a *future* problem that may never occur. Therefore it probably counts as over-engineering.